File: 2019/2.py
base = [int(x) for x in input().split(',')]
for s1 in range(99):
    for s2 in range(99):
        ns = base.copy()
        i = 0   
        ns[1], ns[2] = s1, s2
        while i < len(ns):
            if ns[i] == 1:
                ns[ns[i+3]] = ns[ns[i+1]] + ns[ns[i+2]]
                i += 4
            elif ns[i] == 2:
                ns[ns[i+3]] = ns[ns[i+1]] * ns[ns[i+2]]
                i += 4
            elif ns[i] == 99:
                if s1 == 12 and s2 == 2:
                    print(f"part 1: {ns[0]}")
                if ns[0] == 19690720:
                    print(f"part 2: {s1=}, {s2=} -> {100 * s1 + s2}")
                #print(f"{s1=}, {s2=} -> {ns[0]=}")
                break

File: 2019/2.jl
input = readline("2.txt")
program = [parse(Int32, s) for s in split(input, ",")]
program[2] = 12
program[3] = 2
for i in 1:4:length(program)
    op = program[i]
    if op == 1
        program[program[i+3]+1] = program[program[i+1]+1] + program[program[i+2]+1]
    elseif op == 2
        program[program[i+3]+1] = program[program[i+1]+1] * program[program[i+2]+1]
    elseif op == 99
        break
    end
end
println(program[1])
input = readline("2.txt")
for noun in 0:99
    for verb in 0:99
        local program
        program = [parse(Int32, s) for s in split(input, ",")]
        program[2] = noun
        program[3] = verb
        for i in 1:4:length(program)
            op = program[i]
            if op == 1
                program[program[i+3]+1] = program[program[i+1]+1] + program[program[i+2]+1]
            elseif op == 2
                program[program[i+3]+1] = program[program[i+1]+1] * program[program[i+2]+1]
            elseif op == 99
                break
            end
        end
        if program[1] == 19690720
            println("noun=", noun, "verb=", verb, " ", noun*100+verb)
        end
    end
end